### PR TITLE
Selenium: incorrect determination of baseURL for remote tests

### LINKED ISSUE #11346: Selenium: incorrect determination of baseURL for remote tests
### Describe the bug

In [this example](https://github.com/bvfalcon/primefaces-selenium-baseUrl) is shown, that baseUrl for selenium tests is determined incorrect.
The current algorithm is (implemented in [PrimeSelenium.getUrl(AbstractPrimePage)](https://github.com/primefaces/primefaces/blob/07144423145bdc9b43483158cdc1af9705cfe5f4/primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java#L154)) and incorrect:
1) Why is it necessary to [continue](https://github.com/primefaces/primefaces/blob/07144423145bdc9b43483158cdc1af9705cfe5f4/primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java#L159) with baseUrl determination, if [it was already](https://github.com/primefaces/primefaces/blob/07144423145bdc9b43483158cdc1af9705cfe5f4/primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java#L157) determined?
2) if baseLocation [is null here](https://github.com/primefaces/primefaces/blob/07144423145bdc9b43483158cdc1af9705cfe5f4/primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java#L162),  the string "nullSomethingElse" will be return, that is incorrect.

### Reproducer

_No response_

### Expected behavior

baseUrl for test pages is determined according to following algorithm:
1) get page's getBaseLocation method result. if not null -- baseUrl is found
2) get deploymentAdapter's getBaseUrl method result. if not null -- baseUrl is found
3) get deployment.baseUrl option value. if exists -- baseUrl is found
4) if baseUrl is null -- throw an error.

### PrimeFaces edition

Community

### PrimeFaces version

13.0.5

### Theme

_No response_

### JSF implementation

Mojarra

### JSF version

2.2

### Java version

11

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by bvfalcon**
Fix #11346

### CODE REVIEW

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java [hunk 1]
```diff
@@ -152,13 +156,19 @@ public static void goTo(String partialUrl) {
...
-        DeploymentAdapter deploymentAdapter = ConfigProvider.getInstance().getDeploymentAdapter();
-
         String baseLocation = page.getBaseLocation();
-        if (deploymentAdapter != null) {
+        if (baseLocation == null) {
             baseLocation = getBaseUrl();
         }
-
+        if (baseLocation == null) {
+            StringBuilder messageBuilder = new StringBuilder("Cannot determine base url. Please either configure ").append(DEPLOYMENT_BASEURL).append(" or ");
```
Comment (tandraschko) [inline primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: Shouldn't we just move the RuntimeException to #getBaseUrl?

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/internal/DefaultWebDriverAdapter.java [hunk 2]
```diff
@@ -87,7 +87,7 @@ public WebDriver createWebDriver() {
 
         switch (config.getWebdriverBrowser()) {
             case "firefox":
-                FirefoxOptions firefoxOptions = new FirefoxOptions();
+                FirefoxOptions firefoxOptions = new FirefoxOptions().configureFromEnv();
```
Comment (tandraschko) [inline primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/internal/DefaultWebDriverAdapter.java]: whats that? please create a seperate issue + PR for this

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java [hunk 3]
```diff
@@ -169,7 +179,16 @@ public static String getUrl(AbstractPrimePage page) {
      * @return the full URL
      */
     public static String getUrl(String url) {
-        return getBaseUrl() + url;
+        String baseUrl = getBaseUrl();
+        if (baseUrl == null) {
+            StringBuilder messageBuilder = new StringBuilder("Cannot determine base url. Please either configure ").append(DEPLOYMENT_BASEURL).append(" or ");
```
Comment (tandraschko) [inline primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: please use just normal string here for readability, no StringBuilder

will be optimized by the JVM anyway

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java (hunk 1 above)
Comment (bvfalcon) [reply primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: This is not fully duplication: page's class has method getBaseLocation with same functionality as getBaseUrl. This possibility (to configure baseUrl for each page) could be also noticed in error's message.

**Review by melloware — APPROVED** [review]

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/internal/DefaultWebDriverAdapter.java (hunk 2 above)
Comment (bvfalcon) [reply primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/internal/DefaultWebDriverAdapter.java]: Done

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java (hunk 3 above)
Comment (bvfalcon) [reply primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: Done

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java [hunk 4]
```diff
@@ -152,13 +155,18 @@ public static void goTo(String partialUrl) {
...
-
         String baseLocation = page.getBaseLocation();
-        if (deploymentAdapter != null) {
+        if (baseLocation == null) {
             baseLocation = getBaseUrl();
         }
-
+        if (baseLocation == null) {
+            String message = "Cannot determine base url. Please either configure " + DEPLOYMENT_BASEURL + " or " +
+                    (ConfigProvider.getInstance().getDeploymentAdapter() != null ?
```
Comment (tandraschko) [inline primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: introduce ConfigProvider.getInstance().getDeploymentAdapter() as variable, makes it even more readable
same in the other method

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java [hunk 5]
```diff
@@ -23,6 +23,9 @@
  */
 package org.primefaces.selenium;
 
+import static org.primefaces.selenium.internal.ConfigProvider.DEPLOYMENT_ADAPTER;
```
Comment (tandraschko) [inline primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: in general we avoid static imports in PF

Comment (bvfalcon) [reply primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: Done

File: primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java (hunk 4 above)
Comment (bvfalcon) [reply primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java]: Done

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
looks perfect, thanks!

**Comment by tandraschko** [comment]
@melloware will you cherrypick to 13.0.6?

**Pull request merged**